vicky/build.rs

```diff
@@ -1,3 +1,49 @@
+use std::fs::File;
+use std::io;
+use std::io::{BufRead, BufReader, BufWriter, ErrorKind, Write};
+
 fn main() {
-    println!("cargo:rerun-if-changed=migrations");
-}
\ No newline at end of file
+    if let Err(e) = ensure_locks_id_nullable() {
+        eprintln!("Couldn't update schema.rs: {e}")
+    }
+}
+
+fn ensure_locks_id_nullable() -> io::Result<()> {
+    println!("cargo::rerun-if-changed=src/lib/database/schema.rs");
+    let schema_rs_path = "src/lib/database/schema.rs";
+
+    let schema = File::open(schema_rs_path)?;
+    let reader = BufReader::new(schema);
+    let mut new_schema: String = String::new();
+    let mut found_locks_table = false;
+    let mut only_copy = false;
+
+    for line in reader.lines() {
+        let line = line?;
+
+        if only_copy {
+            new_schema.push_str(&line);
+        } else if !found_locks_table {
+            if line.contains("locks (id) {") {
+                found_locks_table = true;
+            }
+            new_schema.push_str(&line);
+        } else {
+            if !line.contains("id ->") {
```

Comment:
Should we support different primary keys than `id`?

Reply:
ye